otter/util/zk.py

```diff
@@ -170,3 +199,113 @@ def get_zk_dispatcher(kz_client):
             partial(perform_get_stat, kz_client),
         AcquireLock: perform_acquire_lock
     })
+
+
+class PollingLock(object):
+    """
+    Zookeeper lock recipe that polls the children on interval basis instead
+    of leaving a watch on previous child. It's supposed to replace
+    :obj:`kazoo.recipe.lock.Lock` and has same signature and _similar_ behavior
+    for ``acquire`` and ``release`` methods.
+    """
+
+    def __init__(self, dispatcher, path, identifier=""):
+        self.dispatcher = dispatcher
+        self.path = path
+        self.identifier = identifier
+        self._node = None
```

Comment:
> In any case - this is just something I will bring up every time I see a None now, because None is Python's manifestation of the billion dollar mistake and one day I'd like to stop making it.

I think the best way to do this by using a language that does not allow it: Haskell :trollface: